src/Material/Lists.elm

```diff
@@ -0,0 +1,194 @@
+module Material.Lists
+    exposing
+        ( ul
+        , li
+        , twoLine
+        , threeLine
+        , itemPrimaryContent
+        , itemAvatar
+        , itemIcon
+        , itemSecondaryContent
+        , itemSecondaryInfo
+        , itemSecondaryAction
+        , itemTextBody
+        , itemSubTitle
+        )
+
+{-| From the [Material Design Lite documentation](http://www.getmdl.io/components/index.html#lists-section):
+
+> Lists present multiple line items vertically as a single continuous element.
+> Refer the Material Design Spec to know more about the content options.
+
+See also the
+[Material Design Specification]([https://material.google.com/components/lists.html).
+
+Refer to [this site](http://debois.github.io/elm-mdl#/lists)
+for a live demo.
+
+Example use:
+
+    ul []
+        [ li [  ] [  ]
+        , li [  ] [  ]
+        , li [  ] [  ]
+        ]
+
+# Main list elements
+@docs ul, li
+
+# List item configuration for multiple lines
+@docs twoLine, threeLine
+
+# Content configuration for one or multiple lines
+@docs itemPrimaryContent, itemAvatar, itemIcon
+
+# Content configuration for multiple lines
+@docs itemSecondaryContent, itemSecondaryInfo, itemSecondaryAction, itemTextBody, itemSubTitle
+-}
+
+import Platform.Cmd exposing (Cmd, none)
+import Html.Events as Html
+import Html exposing (Html, Attribute)
+import Parts exposing (Indexed)
+import Material.Options as Options exposing (Property, Style, cs, nop)
+
+
+{-| Main list function
+-}
+ul :
+    List (Property {} m)
+    -> List (Html m)
+    -> Html m
+ul options nodes =
+    let
+        summary =
+            Options.collect {} options
+    in
+        Options.apply summary
+            Html.ul
+            [ cs "mdl-list"
+            ]
+            []
+            nodes
+
+
+type ListItemLinesVariant
+    = OneLine
+    | TwoLine
+    | ThreeLine
+
+
+type alias ListItemConfig =
+    { listItemLinesVariant : ListItemLinesVariant
+    }
+
+
+defaultListItemConfig : ListItemConfig
+defaultListItemConfig =
+    { listItemLinesVariant = OneLine
+    }
+
+
+{-| List item should have two lines
+-}
+twoLine : Property { a | listItemLinesVariant : ListItemLinesVariant } m
+twoLine =
+    listItemLines TwoLine
+
+
+{-| List item should have three lines
+-}
+threeLine : Property { a | listItemLinesVariant : ListItemLinesVariant } m
+threeLine =
+    listItemLines ThreeLine
+
+
+{-| List item should have the given number of lines
+-}
+listItemLines : ListItemLinesVariant -> Property { a | listItemLinesVariant : ListItemLinesVariant } m
+listItemLines lines =
+    Options.set <| \self -> { self | listItemLinesVariant = lines }
+
+
+{-| List item
+-}
+li :
+    List (Property ListItemConfig m)
+    -> List (Html m)
+    -> Html m
+li options nodes =
```

Comment:
Maybe make a constructor for each of the three list variants (item, item--two-line, item--three-line)? It would make it easier for user's and it'd be really easy to implement: 

    li options = 
      Options.styled li (cs "mdl-list__item" ++ options)

    li2 options = 
      Options.styled li (cs "mdl-list__item--two-line" ++ options)

Then you can also get rid of the `ListItemLinesVariant`